### PR TITLE
Refactor to separate tables and columns models and view models

### PR DESCRIPTION
**Pull request by sm15455**
Things I have done:

- Removed TableInformationModel
- Moved ShowKeylessWarning and IsProcedure to TableInformationViewModel since they are used only by UI (UI updated to reflect these changes(
- Create IColumnInformationViewModel and ColumnInformationViewModel to handle columns in the UI (UI doesn't handle columns yet)
- Updated view models, dialogs, options class and reverse engineering classes to use TableModel instead of TableInformationModel
- added and .editorconfig file
- In the XAML, I see that checkboxes and text are disabled if the object has no primary key, but the HasPrimaryKey is always set to true. I have removed the possibility to disable chekboxes and text in XAML, but I can undo this change.

Pending:
- Tests must be fixed (don't compile)

The .zip files are not included, they must be recreated.

### CODE REVIEW

File: src/GUI/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,137 @@
+# Rules in this file were initially inferred by Visual Studio IntelliCode from the C:\Work\SMProjects\contributions\EFCorePowerTools\src\GUI codebase based on best match to current usage at 18/11/2020
```
Comment (ErikEJ) [inline src/GUI/.editorconfig:1]: Why is this file included?

File: src/GUI/EFCorePowerTools/Dialogs/PickTablesDialog.xaml [hunk 2]
```diff
@@ -53,11 +53,11 @@
                         <Setter Property="Margin" Value="5, 0, 0, 0"/>
                     </Style>
                     <Style x:Key="TableNameTextBlockStyle" TargetType="TextBlock">
-                        <Style.Triggers>
-                            <DataTrigger Binding="{Binding Model.HasPrimaryKey}" Value="False">
+                        <!--<Style.Triggers>
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools/Dialogs/PickTablesDialog.xaml:56]: Can we just remove this?

File: src/GUI/EFCorePowerTools/Dialogs/PickTablesDialog.xaml [hunk 3]
```diff
@@ -118,16 +118,15 @@
...
                                     d:DataContext="{d:DesignInstance Type=viewModels:TableInformationViewModel, IsDesignTimeCreatable=false}">
                             <CheckBox Style="{StaticResource TableSelectionCheckBoxStyle}"
-                                      IsChecked="{Binding IsSelected}"
-                                      IsEnabled="{Binding Model.HasPrimaryKey}"/>
-                            <TextBlock Text="{Binding Model.Name}" 
+                                      IsChecked="{Binding IsSelected}" />
+                            <TextBlock Text="{Binding Name}" 
                                        Style="{StaticResource TableNameTextBlockStyle}"/>
                             <!--<Image Style="{StaticResource NoPrimaryKeyImageStyle}"
-                                   Visibility="{Binding Model.HasPrimaryKey, Converter={StaticResource BoolToVisibilityConverter}, ConverterParameter=invert}"/>-->
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools/Dialogs/PickTablesDialog.xaml:126]: Remove ?

File: src/GUI/EFCorePowerTools/Handlers/ReverseEngineer/TableListBuilder.cs [hunk 4]
```diff
@@ -39,21 +39,6 @@ public List<TableInformationModel> GetTableDefinitions(bool useEFCore5)
...
-                if (item.IsProcedure)
-                {
-                    item.HasPrimaryKey = true;
-                    continue;
-                }
-
-                if (!item.HasPrimaryKey)
-                {
-                    item.HasPrimaryKey = true;
-                    item.ShowKeylessWarning = true;
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools/Handlers/ReverseEngineer/TableListBuilder.cs:53]: Are views displayed with the icon and warning now?

File: src/GUI/EFCorePowerTools/Handlers/ReverseEngineerHandler.cs [hunk 5]
```diff
@@ -148,12 +148,12 @@ public async Task ReverseEngineerCodeFirstAsync(Project project)
                         preselectedTables.AddRange(normalizedTables);
                     }
 
-                    for (var i = 0; i < predefinedTables.Count; i++)
-                    {
-                        var preselectedTable = preselectedTables.FirstOrDefault(c => c.Name.Equals(predefinedTables[i].Name, StringComparison.OrdinalIgnoreCase));
-                        if (preselectedTable != null)
-                            predefinedTables[i] = preselectedTable;
-                    }
+                    //for (var i = 0; i < predefinedTables.Count; i++)
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools/Handlers/ReverseEngineerHandler.cs:151]: Why is this commented out?

**Comment by sm15455** [comment]
#588

**Review by ErikEJ — CHANGES_REQUESTED** [review]

File: src/GUI/.editorconfig (hunk 1 above)
Comment (sm15455) [reply src/GUI/.editorconfig:1]: My fault. I follow 3 projects (for my customers) and this one. Each project uses different code style. I use this file to adapt to each project style but i shouldn't have committed it. I'll remove it.

**Comment by ErikEJ** [comment]
>> In the XAML, I see that checkboxes and text are disabled if the object has no primary key, but the HasPrimaryKey is always set to true. I have removed the possibility to disable chekboxes and text in XAML, but I can undo this change.

This is a remenant from EF Core 2 support, can be removed

File: src/GUI/EFCorePowerTools/Dialogs/PickTablesDialog.xaml (hunk 2 above)
Comment (sm15455) [reply src/GUI/EFCorePowerTools/Dialogs/PickTablesDialog.xaml:56]: I left it there to discuss with you whether this feature is still needed or not. The HasPrimaryKey was always set to true via code which made impossible to have the check disabled. If this feature is no longer needed I can remove it.

File: src/GUI/EFCorePowerTools/Dialogs/PickTablesDialog.xaml (hunk 3 above)
Comment (sm15455) [reply src/GUI/EFCorePowerTools/Dialogs/PickTablesDialog.xaml:126]: ok

File: src/GUI/EFCorePowerTools/Handlers/ReverseEngineer/TableListBuilder.cs (hunk 4 above)
Comment (sm15455) [reply src/GUI/EFCorePowerTools/Handlers/ReverseEngineer/TableListBuilder.cs:53]: Yes, those fix were needed by the TableInformationViewModel so I moved them there

File: src/GUI/EFCorePowerTools/Handlers/ReverseEngineerHandler.cs (hunk 5 above)
Comment (sm15455) [reply src/GUI/EFCorePowerTools/Handlers/ReverseEngineerHandler.cs:151]: That code is no longer needed because now TableModel object properties are splattered into view model. When the user confirms his selection, the view model properties are used to create new TableModel instances that reflects user selection. Before this change, the TableInformationModel instance was directly used and returned by view models so I had to pass to them the correct TableInformationModel instance (the one coming from db table list or the one coming from preselected table list).

**Comment by sm15455** [comment]
Ok, i'll remove it

File: src/GUI/EFCorePowerTools/Dialogs/PickTablesDialog.xaml (hunk 2 above)
Comment (ErikEJ) [reply src/GUI/EFCorePowerTools/Dialogs/PickTablesDialog.xaml]: You can remove it

File: src/GUI/EFCorePowerTools/Handlers/ReverseEngineerHandler.cs (hunk 5 above)
Comment (ErikEJ) [reply src/GUI/EFCorePowerTools/Handlers/ReverseEngineerHandler.cs:151]: Then why not just delete it?

Comment (sm15455) [reply src/GUI/EFCorePowerTools/Handlers/ReverseEngineerHandler.cs:151]: Good question, my stupidity is the answer. I'll delete it

**Comment by sm15455** [comment]
Great. I'll start working on tests then.
